lib/checks/aria/aria-prohibited-attr-evaluate.js

```diff
@@ -26,42 +26,40 @@ import standards from '../../standards';
  * @memberof checks
  * @return {Boolean} True if the element uses any prohibited ARIA attributes. False otherwise.
  */
-function ariaProhibitedAttrEvaluate(node, options, virtualNode) {
-  const prohibited = [];
-
-  const role = getRole(virtualNode);
-  const attrs = virtualNode.attrNames;
-  const prohibitedAttrs = role
-    ? standards.ariaRoles[role].prohibitedAttrs
-    : ['aria-label', 'aria-labelledby'];
-
-  if (!prohibitedAttrs) {
+function ariaProhibitedAttrEvaluate(node, options = {}, virtualNode) {
+  const { elementsAllowedAriaLabel = [] } = options;
+  const attrBlacklist = getAttrBlacklist(virtualNode, elementsAllowedAriaLabel);
+  if (!attrBlacklist) {
     return false;
   }
-
-  for (let i = 0; i < attrs.length; i++) {
-    const attrName = attrs[i];
+  
+  const prohibited = virtualNode.attrNames.filter(attrName => {
```

Comment:
Instead of looping over all attributes but only caring about the ones that are in the prohibited list, it would be faster to just look at the prohibited list.

(Sorry, multiline suggestion wasn't working for these lines)

```suggestion
  const prohibited = prohibitedAttrs.filter(attrName => {
```